Source/Particles/Gather/FieldGather.H

```diff
@@ -126,6 +132,18 @@ void doGatherShapeN(const amrex::Real * const xp,
                         by_arr(lo.x+j0+ix, lo.y+l0+iz, 0);
                 }
             }
+
+#ifdef WARPX_DIM_RZ
+            // Convert Exp and Eyp (which are actually Erp and Ethetap) to Ex and Ey
+            const amrex::Real theta = std::atan2(yp[ip], xp[ip]);
+            const amrex::Real Exp_save = Exp[ip];
+            Exp[ip] = std::cos(theta)*Exp[ip] - std::sin(theta)*Eyp[ip];
+            Eyp[ip] = std::cos(theta)*Eyp[ip] + std::sin(theta)*Exp_save;
+            const amrex::Real Bxp_save = Bxp[ip];
+            Bxp[ip] = std::cos(theta)*Bxp[ip] - std::sin(theta)*Byp[ip];
+            Byp[ip] = std::cos(theta)*Byp[ip] + std::sin(theta)*Bxp_save;
```

Comment:
Same comment as before: we could avoid calculating `atan` and `sin`/`cos` by using `xp[ip]/r` and `yp[ip]/r`.